### PR TITLE
Framework Refactor: episode 1 split files and move into framework package

### PR DESCRIPTION
**Pull request by shiywang**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by shiywang** [comment]
/assign @lukas-bednar 

please also see my comments [here.](https://github.com/kubevirt/kubevirt-ansible/pull/411#issuecomment-440112022)

**Comment by lukas-bednar** [comment]
This change looks good to me. I think it is good start .

File: tests/framework/obj_dump.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+			"Failed to login as Admin user. Skipping obj desc collection",
+		)
+		return err
+	}
+
+	fmt.Println("Running log collection")
+	err = CollectObjDesc(dest, "pod", "pv", "pvc")
+
+	if err != nil {
+		fmt.Printf("Failed to collect logs\n%s", err)
```
[the remark this answers is not in the document]
Comment (shiywang) [reply tests/framework/obj_dump.go]: @gbenhaim I think this is a warning, we can add `Warning` at the beginning.

File: tests/framework/namespace.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+func RemoveNamespaces() {
+	virtCli, err := kubecli.GetKubevirtClient()
+	ktests.PanicOnError(err)
+	testNamespaces := []string{ktests.NamespaceTestDefault, ktests.NamespaceTestAlternative}
+
+	// First send an initial delete to every namespace
+	for _, namespace := range testNamespaces {
+		err := virtCli.CoreV1().Namespaces().Delete(namespace, nil)
+		if !errors.IsNotFound(err) {
+			ktests.PanicOnError(err)
```
Comment (nellyc) [inline tests/framework/namespace.go]: the usage of PanicOnError is problematic, the execution will die and you end up with nothing. this is especially wrong to do when the execution is about to finish, we will lose all tests results

File: tests/framework/cli_exec.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	filePath      string
+	nameSpace     string
+	query         string
+	expectOut     string
+	actualOut     string
+	params        []string
+}
+
+func execute(r Result) string {
+	return executeWithCustomTimeout(r, LongTimeout)
```
Comment (nellyc) [inline tests/framework/cli_exec.go]: why LongTimeout as default?

Comment (shiywang) [reply tests/framework/cli_exec.go]: my previous thought is user specify their own timeout value case by case, but anyway I can change it to a more proper way

File: tests/framework/namespace.go (hunk 2 above)
Comment (shiywang) [reply tests/framework/namespace.go]: could you give me an example how do we report errors ?

**Comment by shiywang** [comment]
ci test please

**Comment by shiywang** [comment]
ci test please

**Pull request merged**